### PR TITLE
#14 Logging

### PR DESCRIPTION
**Pull request by nils-herrmann**
New PR to avoid confusion due to merge conflicts and commit reversions.

Log all queries made with`ScopusSearch()` and `AuthorSearch()` at `DEBUG` level in `~/.cache/sosia/sosia.log`

### CODE REVIEW

File: sosia/utils/logger.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""Module to configure the logging for the package."""
+import logging
+from pathlib import Path
+
+_cache_folder = Path.home() / '.cache' / 'sosia/'
```
Comment (Michael-E-Rose) [inline sosia/utils/logger.py]: Can we name it `LOG_FILE` please and include the full file name? Will make it easier for me to put all filepaths in a separate file later on.

File: sosia/utils/logger.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+_cache_folder.mkdir(parents=True, exist_ok=True)
+
+logger = logging.getLogger('sosia')
+
+file_handler = logging.FileHandler(_cache_folder / 'sosia.log', mode='a', encoding='utf-8')
```
Comment (Michael-E-Rose) [inline sosia/utils/logger.py]: If this is append mode, will it always write to a new file? Ideally I want a fresh file when I initiate a new `Original()`. It's okay when it overwrites the old stuff, but it's not ok when the file grows forever.

**Comment by Michael-E-Rose** [comment]
That's a good start already. But we need to write it such that the file comes out fresh when I initiate a new `Original()`. Maybe we put the creation of the logfile in there, and move the log file connection around as private attribute?

And finally two requests on the stored information:
1. We need to store the view of the request.
2. As some queries can become very long (the max query string length is 255 characters), I think it's better to have it a separate line.
3. Maybe we can log each query like so: Timestamp, Scopus API (or pybliometrics class used) and view, query, length of results. Each of these information on a separate line, ended by newline.

**Comment by nils-herrmann** [comment]
The `LOG_FILE.log` now has entries like:
```log
24-11-05 21:04:40 - DEBUG: 
	- Scopus class: AuthorSearch
	- View: STANDARD
	- Query: AU-ID(53164702100)
	- Nr Results: 1
24-11-05 21:04:41 - DEBUG: 
	- Scopus class: ScopusSearch
	- View: COMPLETE
	- Query: REF(55208373700 OR 55208373700) AND PUBYEAR BEF 2017 AND NOT AU-ID(55208373700) AND NOT AU-ID(55208373700)
	- Nr Results: 0
24-11-05 21:04:42 - DEBUG: 
	- Scopus class: ScopusSearch
	- View: COMPLETE
	- Query: REF(2-s2.0-84959420483 OR 2-s2.0-84949113230) AND PUBYEAR BEF 2017 AND NOT AU-ID(55208373700) AND NOT AU-ID(55208373700)
	- Nr Results: 0
```

**Comment by Michael-E-Rose** [comment]
Yeah, that looks great! Let's just change some words here and then we're done: 
- Scopus class -> Scopus API
- try to have a blank in the class name, so it's really the API and not the class
- Nr Results -> Results

**Comment by nils-herrmann** [comment]
A log entry now looks like this:
```log
24-11-11 09:59:52 - DEBUG: 
	- Scopus API: Scopus Search
	- View: COMPLETE
	- Query: SOURCE-ID(14204) AND PUBYEAR IS 1984
	- Results: 730
```

**Comment by Michael-E-Rose** [comment]
Last request: Can you please move all the new functions from `sosia/utils/logger.py` to `sosia/establishing/logger.py`? (And then remove the new `utils` folder again.)

**Comment by nils-herrmann** [comment]
I moved the file to the `establishing` folder (and changed the corresponding imports).

What do you mean by this?
> (And then remove the new utils folder again.)

**Comment by Michael-E-Rose** [comment]
> What do you mean by this?
> > (And then remove the new utils folder again.)

You created a new `utils` folder. But you removed it already, so all is fine. Thanks, Nils!

**Pull request merged**